src/accessibility/describe.js

```diff
@@ -0,0 +1,304 @@
+ * description accessible to screen-reader users only, in
+ * <a href="https://developer.mozilla.org/en-US/docs/Web/API/Canvas_API/Tutorial/Hit_regions_and_accessibility" target="_blank">
+ * a sub DOM inside the canvas element</a>. If a second parameter is not
+ * specified, by default, the description will only be available to
+ * screen-reader users.
+ *
+ * @method describe
+ * @param  {String} text      description of the canvas
+ * @param  {Constant} [display] either LABEL or FALLBACK (Optional)
+ *
+ * @example
+ * <div>
+ * <code>
+ * describe('pink square with red heart in the bottom right corner', LABEL);
+ * background('pink');
+ * fill('red');
+ * noStroke();
+ * ellipse(67, 67, 20, 20);
+ * ellipse(83, 67, 20, 20);
+ * triangle(91, 73, 75, 95, 59, 73);
+ * </code>
+ * </div>
+ *
+ * <div>
+ * <code>
+ * let x = 0;
+ * function draw() {
+ *   if (x > 100) {
+ *     x = 0;
+ *   }
+ *   background(220);
+ *   fill(0, 255, 0);
+ *   ellipse(x, 50, 40, 40);
+ *   x = x + 0.1;
+ *   describe('green circle at x pos ' + round(x) + ' moving to the right');
+ * }
+ * </code>
+ * </div>
+ *
+ */
+
+p5.prototype.describe = function(t, d) {
+  const cnvId = this.canvas.id;
+  p5._validateParameters('describe', arguments);
+  //Creates a sub DOM inside of the canvas element and populates
+  //it with description text.
+  if (typeof t !== 'string') {
+    return;
+  }
+  t = this._descriptionText(t);
+  if (document.getElementById(cnvId + ds) === null) {
+    document.getElementById(cnvId).innerHTML =
+      '<div id="' +
+      cnvId +
+      ds +
+      '" role="region" aria-label="Canvas Description"><p id="' +
+      cnvId +
+      fds +
+      '"></p></div>';
+  } else if (document.getElementById(cnvId + fds) === null) {
+    document
+      .getElementById(cnvId + ft)
+      .insertAdjacentHTML('beforebegin', '<p id="' + cnvId + fds + '"></p>');
+  }
+  if (document.getElementById(cnvId + fds).innerHTML !== t) {
+    document.getElementById(cnvId + fds).innerHTML = t;
+  }
+  //If display is LABEL creates a div adjacent to the canvas element with
+  //description text.
+  if (d === this.LABEL) {
+    if (document.getElementById(cnvId + lb) === null) {
+      document
+        .getElementById(cnvId)
+        .insertAdjacentHTML(
+          'afterend',
+          '<div id="' +
+            cnvId +
+            lb +
+            '" class="p5Label"><p id=' +
+            cnvId +
+            lds +
+            '></p></div>'
+        );
+    } else if (document.getElementById(cnvId + lds) === null) {
+      document
+        .getElementById(cnvId + lt)
+        .insertAdjacentHTML('beforebegin', '<p id=' + cnvId + lds + '></p>');
+    }
+    if (document.getElementById(cnvId + lds).innerHTML !== t) {
+      document.getElementById(cnvId + lds).innerHTML = t;
+    }
+  }
+};
+
+/**
+ * Helper function for describe() and describeElement().
+ */
+
+p5.prototype._descriptionText = function(t) {
```

Comment:
if this is only used internally, it doesn't need to be attached to `p5.prototype`